From 329a3b151068caee84e10db82cecf5946a6b676a Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Tue, 14 Oct 2014 13:18:55 -0700
Subject: [PATCH 4957/5965] PD #98446. Move ionvideo canvas define to canvas.h,
 change the canvas use for ionvideo to avoid conflict with deinterlace driver.

Change-Id: I8b655586706f2df753d60c999ed948dcaebed5e3
---
 drivers/amlogic/ionvideo/ionvideo.h    | 3 ---
 include/linux/amlogic/amports/canvas.h | 3 +++
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/ionvideo/ionvideo.h b/drivers/amlogic/ionvideo/ionvideo.h
index e58613c02a1e..2ac690a8828d 100755
--- a/drivers/amlogic/ionvideo/ionvideo.h
+++ b/drivers/amlogic/ionvideo/ionvideo.h
@@ -42,9 +42,6 @@
 #define MAX_WIDTH 1920
 #define MAX_HEIGHT 1088
 
-#define PPMGR2_MAX_CANVAS 10
-#define PPMGR2_CANVAS_INDEX 0x66
-
 #define DUR2PTS(x) ((x) - ((x) >> 4))
 
 #define dprintk(dev, level, fmt, arg...)                    \
diff --git a/include/linux/amlogic/amports/canvas.h b/include/linux/amlogic/amports/canvas.h
index 08082df65f97..0116f3fc3b6d 100755
--- a/include/linux/amlogic/amports/canvas.h
+++ b/include/linux/amlogic/amports/canvas.h
@@ -40,6 +40,9 @@ typedef struct {
 #define AMVDEC_ALL_CANVAS_RANGE_0 0xb  // vc1/real/mpeg12/mpeg4/ jpeg logo / h264
 #define AMVDEC_ALL_CANVAS_RANGE_1 0x25 //mjpeg
 
+#define PPMGR2_MAX_CANVAS 10
+#define PPMGR2_CANVAS_INDEX 0x2c    // 0x2c-0x3f for PPMGR2 (IONVIDEO)
+
 #define AMVDEC_H264MVC_CANVAS_INDEX 0x78
 #define AMVDEC_H264MVC_CANVAS_MAX 0xbf
 
-- 
2.19.0

